discovery_engine/lib/src/domain/event_handler.dart

```diff
@@ -342,3 +356,14 @@ class EventHandler {
     }
   }
 }
+
+/// Thrown when the engine cannot be recovered from the given state.
+class InvalidEngineStateException implements Exception {
+  final String message;
+
+  InvalidEngineStateException(this.message);
+
+  @override
+  String toString() =>
+      'InvalidEngineStateException: $message. Try to restart the engine.';
```

Comment:
> Try to restart the engine.

Do you think it makes sense for us to try again after clearing state? or is should be tried in the Manager, or the app side?

Reply:
Yes we could do that. the reason why we don't do it is that we want to let team blue log the error/event.